src/client/fsharp/Oak/GraphView.fs

```diff
@@ -166,43 +165,90 @@ let view =
+                    let scaleValue =
                         match model.GraphViewOptions.Scale with
                         | NoScale -> 1
-                        | SubTreeNodes -> if not n.Limited then 1 else n.Size
-                        | AllNodes -> n.Size })
-
-            let edges =
+                        | SubTreeNodes ->
+                            if not graphOakNode.Limited then
+                                1
+                            else
+                                graphOakNode.Size
+                        | AllNodes -> graphOakNode.Size
+
+                    {| id = !!graphOakNode.Id
+                       label = graphOakNode.Node.Trim()
+                       level = graphOakNode.Level
+                       color = getColor graphOakNode.Type
+                       shape = if graphOakNode.Limited then "box" else "ellipse"
+                       value = scaleValue
+                       font = {| color = getFontColor graphOakNode.Type |} |})
+
+            let edges: VisNetwork.edge array =
                 oakNodes
                 |> Map.values
                 |> Seq.collect (fun n ->
                     n.Children
                     |> Seq.map (fun m ->
                         if m.Type = Standard then
-                            { From = NodeId n.Id
-                              To = NodeId m.Id
-                              Dashed = false }
+                            {| from = !!n.Id
+                               ``to`` = !!m.Id
+                               dashes = false |}
+                            : VisNetwork.edge
                         else
-                            { From = NodeId m.Id
-                              To = NodeId n.Id
-                              Dashed = true }))
-                |> set
+                            {| from = !!m.Id
+                               ``to`` = !!n.Id
+                               dashes = true |}))
+                |> Seq.toArray
+
+            let layout =
+                let hier =
+                    {| enabled = true
+                       direction = "UD"
+                       levelSeparation = 75 |}
+
+                match model.GraphViewOptions.Layout with
+                | TopDown -> {| hierarchical = hier |}
+                | LeftRight -> {| hierarchical = {| hier with direction = "LR" |} |}
+                | Free -> {| hierarchical = {| hier with enabled = false |} |}
+
+            let parentElement = Browser.Dom.document.getElementById "tab-content"
+
+            let options: VisNetwork.options =
+                {| layout = layout
+                   interaction = {| hover = true |}
+                   width = $"{parentElement.clientWidth}"
+                   height = $"{parentElement.clientHeight}"
+                   nodes =
+                    {| scaling =
+                        {| label = scalingLabel
+                           min = minScaling |} |} |}
```

Comment:
Removing `min` here solve scaling issue for me.

Reply:
Hah, interesting. I found that I couldn't see anything without the min.
`minScaling` is used in two locations. Did you remove them both?

Reply:
Just this one. Without second one its not working (tiny nodes).

Not sure what you mean, in old version `min` is not used.

But scaling still working weirdly, nodes are scaled up if you delve down to the tree (no collapsed node, but all nodes are big) or sometimes when change to free mode.

Probably scaling work differently here, and it need some experimenting with values.

Reply:
Yes, I believe the settings are slightly different now.
`vis-react` was using `vis 4.x` while `vis-network` is `9.1.x` so we are probably trying to update a few majors.